### PR TITLE
Update tree.go

### PR DESCRIPTION
**Pull request by badis**
Fixed typos in comments

### CODE REVIEW

File: pkg/tree/tree.go
Comment (mateiidavid) [inline pkg/tree/tree.go:137]: I'm not sure this is a typo. I think the intent with "iff" is to say "if and only if". I think the distinction here is logical. When the tree is empty, it is necessary that the tree contains no leaf values. But without the `iff` distinction, it is not necessary for no leaf values to guarantee an empty tree (there could be other requirements for a tree to be empty, not just zero leaf values).

That's why we have this equivalence here, we want the statement to be bidirectional:
* If a tree is empty, it has no leaf values.
* If a tree has no leaf values, it is empty.

a tree is empty iff it contains no leaf values.

At least, this is how I read it. I hope it makes sense; for the purpose of clarity, I'd leave this as is :)

File: pkg/tree/tree.go
Comment (mateiidavid) [inline pkg/tree/tree.go:151]: We tend to wrap our comments at col 80. I think this was split into multiple lines because of that (it's less than 80 characters on that line but it wouldn't make sense to split "Tree").

**Review by mateiidavid — COMMENTED** [review]
Thank you for your work @badis! Cleaning up code and comments is always very appreciated in my book. I've left my thoughts on the PR. I'm not sure if I'd categorise these as typos necessarily, but please wait for another maintainer to review. :)

File: pkg/tree/tree.go
Comment (badis) [reply pkg/tree/tree.go:137]: Thank you very much @mateiidavid. Yes, it makes sense to me now.

File: pkg/tree/tree.go
Comment (badis) [reply pkg/tree/tree.go:151]: Ok, I see the fact that code should be wrapped at 80 chars. but still there is a duplication of `as`

**Review by adleong — APPROVED** [review]

**Review by kleimkuhler — APPROVED** [review]

**Pull request merged**
